File: graph-traversals/bfs.java
void bfs(Map<Integer, List<Integer>> graph, int node) {
    Deque<Integer> q = new ArrayDeque<>();
    Set<Integer> visited = new HashSet<>();
    q.addLast(node);
    visited.add(node);

    while (q.isEmpty() == false) {
        int cur = q.removeFirst();
        System.out.print(cur + " ");
        for (int next : graph.get(cur)) {
            if (visited.contains(next)) continue;
            q.addLast(next);
            visited.add(next);
        }
    }
}

File: graph-traversals/bfs.py
def bfs(graph: Dict[int, List[int]], node: int):
    q = deque([node])
    visited = set([node])

    while q:
        cur = q.popleft()
        
        print(cur, end = " ")
        
        for next in graph[cur]:
            if next in visited: 
                continue
                
            q.append(next)
            visited.add(next)